observatory-platform/observatory/platform/cli/cli.py

```diff
@@ -222,17 +222,26 @@ def generate():
 
 
 @generate.command()
-@click.argument("telescope_type", nargs=1)
-@click.argument("telescope_name", nargs=1)
-def telescope(telescope_type: str, telescope_name: str):
-    """Generate boiler plate code for a new telescope.
-
-    telescope_type: Type of telescope. Options are Telescope, StreamTelescope, SnapshotTelescope.
-    telescope_name: Name of your new telescope.
+@click.argument(
+    "project_path", type=click.Path(exists=True, file_okay=False, dir_okay=True),
+)
+@click.argument("package_name", type=str)
+@click.argument("workflow_type", type=click.Choice(["Workflow", "StreamTelescope", "SnapshotTelescope"]))
+@click.argument("workflow_name", type=str)
+def workflow(project_path: str, package_name: str, workflow_type: str, workflow_name: str):
+    """Generate boiler plate code for a new workflow.
+
+    project_path: The Python project path.
+    package_name: The Python package name.
+    workflow_type: Type of workflow. Options are Workflow, StreamTelescope, SnapshotTelescope.
+    workflow_name: Name of your new workflow.
```

Comment:
Might be good to give an example of usage, and the summary of the directory structure it produces.

Reply:
This will be covered by the changes in the telescope documentation branch, at least that is what I'm assuming, because there are still quite a lot of difference in the `generate workflow` command for these two branches.

Reply:
I've added a bit more help text, I think project path and package name will disappear after your changes Aniek:
```
Usage: observatory generate workflow [OPTIONS] PROJECT_PATH PACKAGE_NAME [Work
                                     flow|StreamTelescope|SnapshotTelescope]
                                     WORKFLOW_NAME

  Generate a new workflow.

  PROJECT_PATH is the Python project path.
  PACKAGE_NAME is the Python package name.
  WORKFLOW_TYPE is the type of workflow.
  WORKFLOW_NAME is the the name of your new workflow.

  For example, the command: observatory generate workflow /path/to/my-workflows-project my_workflows_project Workflow MyWorkflow

  Will generate the following files and folders:

  └── my-workflows-project
      ├── docs
      │   ├── index.rst
      │   └── my_workflow.md
      ├── my_workflows_project
      │   ├── dags
      │   │   ├── __init__.py
      │   │   └── my_workflow.py
      │   ├── __init__.py
      │   └── workflows
      │       ├── __init__.py
      │       └── my_workflow.py
      └── tests
          └── my_workflows_project
              ├── __init__.py
              └── workflows
                  ├── __init__.py
                  └── test_my_workflow.py

Options:
  --help  Show this message and exit.

```

To generate that example folder structure, you can run the generate command and then install the program `tree` and then run tree from the folder above `my-workflows-project` and copy paste.

Reply:
Yeah do you mean assuming that the command is run from inside a project?

Also, is there a need to have the package_name as a subdirectory in the tests directory?
Could it be `tests/workflows/test_my_workflow.py`?